.github/workflows/label_when_reviewed.yml

```diff
@@ -17,7 +17,9 @@
 #
 ---
 name: Label when reviewed
-on: pull_request_review  # yamllint disable-line rule:truthy
+on:  # yamllint disable-line rule:truthy
+  pull_request_review:
+    types: [submitted]
```

Comment:
Why do we want to remove `dismissed` or `edited` ? 

For sure at least  dismissed for sure can change the status of the review, so it should be there (for example if the request was not approved by one reviewer and approved by another, dismissing the 'non-approved' status will change overall status of the PR to "approved". So we definitely need this one. 

I am not sure about `edited`, but I believe this is the same (documentation is quite vague about it). you can change the state of your review from "request changes"  to "approved" and I believe this is the trigger for the event.

Note that this is not "pull_request_review_comment" type of event. This is a totally different event and we have no workflow for it https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#pull_request_review_comment

From my observations 'pull_request_review' even is NOT triggered every time you make a comment, if this is what you are trying to prevent. But I might be wrong regarding the 'edit' behaviour.